Dockerfile

```diff
@@ -1,34 +1,22 @@
 # Build the manager binary
-FROM golang:1.16 as builder
+FROM golang:1.17 as builder
 
 WORKDIR /workspace
-# Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
 RUN go mod download
 
-# Copy the go source
 COPY main.go main.go
 COPY api/ api/
 COPY controllers/ controllers/
-COPY utils/ utils/
```

Comment:
Were they being used anywhere else (other than `controller`)?

Reply:
controller and util folders are completely different entities and I dont see any reason to put util inside controller folder

Reply:
They are completely different entities because they were inside separate packages for no reason, since they are only being used in one part of the codebase. If this was not the case, we could have it inside `pkg` (as the go standard layout specifies: https://github.com/golang-standards/project-layout#pkg), or `internal` if it was repo-wide and not for use by external packages.

Reply:
However this is not the case, and logically, at this point in time, those packages are moved to where they are needed.

Reply:
@rexagod  I don't think it is good practice to have packages inside packages if it must and the codebase is huge, 
here we have not more than 4 folders and I don't see any problem putting under in the main folder,

Reply:
If we do this now we'll have to refactor later. I'm not doing any of this from my own, I'm just going by what's defined in the standard.

Reply:
Resolved over call: move templates to project root.